ts/sagas/startup/loadServiceDetailRequestHandler.ts

```diff
@@ -100,3 +106,109 @@ export function* watchServicesDetailLoadSaga(
     );
   }
 }
+
+/**
+ * listen for loading services details events to extract some track information
+ * like amount of details to load and how much time they take
+ */
+function* watchLoadServicesDetailToTrack() {
+  yield takeLatest(
+    [loadServicesDetail, loadServiceDetail.success, applicationChangeState],
+    action => {
+      switch (action.type) {
+        // request to load a set of services detail
+        // copying object is needed to avoid "immutable" error on frozen objects
+        case getType(loadServicesDetail):
+          const stats: ServicesDetailLoadTrack = {
+            ...servicesDetailLoadTrack,
+            kind: undefined,
+            startTime: new Date().getTime() as Millisecond,
+            servicesId: new Set([...action.payload]),
+            loaded: 0,
+            toLoad: action.payload.length
+          };
+          servicesDetailLoadTrack = stats;
+          break;
+        // single service detail is been loaded
+        case getType(loadServiceDetail.success):
+          servicesDetailLoadTrack.servicesId.delete(action.payload.service_id);
+          const statsServiceLoad: ServicesDetailLoadTrack = {
+            ...servicesDetailLoadTrack,
+            loaded:
+              servicesDetailLoadTrack.toLoad -
+              servicesDetailLoadTrack.servicesId.size
+          };
+          servicesDetailLoadTrack = statsServiceLoad;
+          if (statsServiceLoad.servicesId.size === 0) {
+            // all service are been loaded
+            trackServicesDetailLoad({
+              ...servicesDetailLoadTrack,
+              kind: "COMPLETE",
+              loadingTime: (new Date().getTime() -
+                servicesDetailLoadTrack.startTime) as Millisecond
+            });
```

Comment:
With some multiple tries on Android, I ran into this case:
![Schermata 2021-02-05 alle 15 07 18](https://user-images.githubusercontent.com/26501317/107043865-d8e37100-67c3-11eb-8f02-b6448b4b3f20.png)

My proposal:
create a common function to calculate the loading time

```typescript
const calculateLoadingTime = (startTime: Millisecond): Millisecond =>
  (startTime !== 0 ? new Date().getTime() - startTime : 0) as Millisecond;
```

and use it at line `147` and `163`.

Reply:
Maybe this scenario is when the app changes state but actually it is not loading any service
I add your suggestion and a condition to send `"PARTIAL"` only when there is a services loading running

https://github.com/pagopa/io-app/pull/2785/commits/16b148168d7222de6d2d4cb56a653258d4ccc6cd